src/AstPragma.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2017 Souffle Developers
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstPragma.h
+ *
+ * Define the class AstPragma to update global options based on parameter.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "AstNode.h"
+#include "AstTransformer.h"
+
+#include <string>
+
+namespace souffle {
+
+/**
+ * @class AstPragma
+ * @brief Intermediate representation of an argument of a Literal (e.g., a variable or a constant)
```

Comment:
Still has the AstArgument description. Maybe something like, "Representation of a global option"